optaweb-vehicle-routing-backend/src/test/java/org/optaweb/vehiclerouting/plugin/planner/RouteOptimizerImplTest.java

```diff
@@ -222,58 +135,145 @@ void removing_wrong_vehicle_should_fail_fast() {
+                .isThrownBy(() -> routeOptimizer.removeLocation(location3))
+                .withMessageContaining("exist");
+    }
+
     @Test
     void added_vehicle_should_be_moved_to_the_depot_even_if_solver_is_not_yet_solving() {
         // arrange
-        final long vehicleId = 10;
+        // -- depot
         routeOptimizer.addLocation(location1, distanceMatrix);
-        clearInvocations(eventPublisher);
+        // -- vehicles
+        routeOptimizer.addVehicle(vehicle(7));
+        routeOptimizer.addVehicle(vehicle(8));
 
         // act
-        routeOptimizer.addVehicle(vehicle(vehicleId));
+        // -- first visit
+        routeOptimizer.addLocation(location2, distanceMatrix);
 
         // assert
-        assertThat(solver.isSolving()).isFalse(); // with a depot and no visits, solver is not yet solving
-        // Can't exactly verify that vehicle.depot == depot but without that, even publishing would fail.
-        RouteChangedEvent event = verifyAndCaptureEvent();
-        assertThat(event.vehicleIds()).containsExactly(vehicleId);
-        assertThat(event.depotId()).isNotEmpty();
-        assertThat(event.routes()).hasSize(1);
+        VehicleRoutingSolution solution = verifySolverStartedWithSolution();
+        assertThat(solution.getVehicleList())
+                .hasSize(2)
+                .allMatch(vehicle -> vehicle.getDepot().getId().equals(location1.id()));
+
+        assertThat(solution.getDepotList()).isNotEmpty();
+        assertThat(solution.getCustomerList()).isNotEmpty();
     }
 
     @Test
-    void solver_should_start_when_two_locations_added() {
-        // add 2 locations
+    void solver_should_start_when_vehicle_is_added_and_there_is_at_least_one_visit() {
+        // arrange
+        routeOptimizer.addLocation(location1, distanceMatrix);
+        routeOptimizer.addLocation(location2, distanceMatrix);
+        verifyZeroInteractions(solverManager);
+
+        // act
+        routeOptimizer.addVehicle(vehicle(9));
+
+        // assert
+        VehicleRoutingSolution solution = verifySolverStartedWithSolution();
+        assertThat(solution.getVehicleList()).hasSize(1);
+        assertThat(solution.getCustomerList()).hasSize(1);
+    }
+
+    @Test
+    void each_location_should_have_a_distance_map_after_it_is_added() {
+        Map<Long, Double> distanceMap = new HashMap<>(1);
+        double distance = 8.079;
+        distanceMap.put(location2.id(), distance);
+        when(distanceMatrix.getRow(location1)).thenReturn(distanceMap);
+        routeOptimizer.addLocation(location1, distanceMatrix);
+
+        verify(distanceMatrix).getRow(location1);
+        VehicleRoutingSolution solution = verifyPublishingPreliminarySolution();
+        assertThat(solution.getDepotList()).hasSize(1);
+        assertThat(solution.getDepotList().get(0).getLocation().getDistanceTo(LocationFactory.fromDomain(location2)))
+                .isEqualTo(8);
+        //FIXME should be: .isEqualTo(distance);
```

Comment:
Do you plan to address this in some next PR?

Reply:
Yes this is related to `optaplanner-examples` where distances are rounded. I'll fix this when forking the planning domain.